### ISSUE TITLE
Elixir generator `AnyType` support

### ISSUE THREAD

**Issue by paulbalomiri** [OpenAPITools/openapi-generator#14070]
##### Description
When compiling spec files which contain `AnyType`, the elixir generator expects a `:struct` type to be used in `decoder.ex`.

The generated code fails because  no type `<PKG>.Model.AnyType` exists

Example Generated code found [here](https://github.com/ory/sdk/blob/847fa4451dd67b624a2820402e92ec6385128921/clients/hydra/elixir/lib/ory/model/o_auth2_client.ex#L112-L113):
```elixir
defimpl Poison.Decoder, for: Ory.Model.OAuth2Client do
  import Ory.Deserializer
  def decode(value, options) do
    value
    |> deserialize(:jwks, :struct, Ory.Model.AnyType, options)
    |> deserialize(:metadata, :struct, Ory.Model.AnyType, options)
  end
end
```

##### openapi-generator version
`6.0.3` to latest in `main` branch

##### Related issues/PRs

The proposed fix will reference this issue and builds on the latest work by @halostatue.

* #12731 
* #12775

I  also include the related elixir issues from #12731.

* #12484
* #4011
Still there is no elixir test suite
*  #8804
* #12268

##### Suggest a fix/enhancement

When work on my branch is finished I intend to submit a PR

**Comment by paulbalomiri** [comment]
I'm referring here to the discussion in [the PR](https://github.com/OpenAPITools/openapi-generator/pull/14071#discussion_r1027254081), which is better placed here, and in separate (future) related issues.

So there are 2 bugs here, and potentially a discussion 
 

1. recreate a PR branch by cherry picking: 
    * for the `AnyType` specific changes (mapping to the bare json decoded value) 

This is a feature addition, but without it this is a bug, as the _non-existing_  `AnyType` is used (as `<PKG>.Model.AnyType`  ) in the generated code:

https://github.com/ory/sdk/blob/958314d130922ad6f20f439b5230141a832231a5/clients/kratos/elixir/lib/ory/model/error_authenticator_assurance_level_not_satisfied.ex#L36-L41

    * change the header field name handling using some form of `Kernel.to_string/1`, as `Tesla.put_header/3` ([doc](https://hexdocs.pm/tesla/Tesla.html#put_header/3)) expets a `binary()` not an `atom()` as `key` argument, as is proposed here:   
  https://github.com/OpenAPITools/openapi-generator/blob/f0625b83fbbb90036d6ce983fa738274aedbd625/samples/client/petstore/elixir/lib/openapi_petstore/request_builder.ex#L105-L108

2. create a separate issue for the tesla/piping part, which is more a discussion whether there is a place for this approach in openapi-generator, in an additional generator, or in a separate custom template repo.

**Comment by paulbalomiri** [comment]
@mrmstn so the two bugs in one PR, discussion separately is ok?

**Comment by paulbalomiri** [comment]
Will contribute this this weekend. [personal urgency]

**Comment by parkerduckworth** [comment]
Hello 👋 any update on this, by any chance? I'm running into the same issue with this elixir kratos client

**Comment by paulbalomiri** [comment]
Hi,
No update for this issue, sorry. It's still on my list "sonnish" :)
In the meantime I'm using the `bugfix_elixir` branch from [my fork](https://github.com/paulbalomiri/ory_sdk/tree/bugfix_elixir) dependency like so:
```elixir
{:ory_hydra, github: "paulbalomiri/ory_sdk", sparse: "clients/hydra/elixir/", tag: "bugfix_elixir"},
{:ory_kratos, github: "paulbalomiri/ory_sdk", sparse: "clients/kratos/elixir/", tag: "bugfix_elixir"},
```

You could fork yourself, or use it directly, but it's not production grade.

**Comment by mrmstn** [comment]
Hi everyone, I was able to get some spare time to resolve this issue.

Thanks to everyone involved in this, specially @paulbalomiri and for bringing this issue up and filling a draft for the issue. Also, a special thanks to @halostatue for the reviews and discussions, I highly appreciate it.

The fix will be shipped in version 6.3.0.
